src/test_case.ts

```diff
@@ -25,37 +25,64 @@ export class TestCase {
     if (this.plan.hasOwnProperty("suites") === false) {
       return;
     }
+
+    // Track the execution of hooks
+    let executedBeforeAllSuiteHook = false;
+    let executedAfterAllSuiteHook = false;
+
     Object.keys(this.plan.suites).forEach((suiteName) => {
+      // Track the execution of hooks
+      let executedBeforeEachSuiteHook = false;
+      let executedAfterEachSuiteHook = false;
+      let executedBeforeAllCaseHook = false;
+      let executedAfterAllCaseHook = false;
+
```

Comment:
catching up... why all these flags? and why did it need to be fast-tracked (1 approval)?

Reply:
it's because of the way deno test works. deno test queues up the test cases before running them and since we're bound to deno test, we need to put the hooks in the test cases. if we don't, then all of the hooks will run before any tests. since the hooks are in the test case, the only way i could see the hooks working properly is with these flags.

on the note about 1 approval. i've only ever required 1 approval. should we change to a 2 approval system?